mlir/include/mlir/Dialect/ArmSME/Transforms/Passes.td

```diff
@@ -116,10 +116,14 @@ def EnableArmStreaming
                             "not be used for input and/or output and the "
                             "function must return with ZA unchanged")
            )}]>,
-    Option<"onlyIfRequiredByOps", "only-if-required-by-ops", "bool",
+    Option<"ifRequiredByOps", "if-required-by-ops", "bool",
            /*default=*/"false",
-           "Only apply the selected streaming/ZA modes if the function "
-           " contains ops that require them.">
+           "Apply the selected streaming/ZA modes if the function contains ops "
+           "that require them.">,
+    Option<"ifContainsScalableVectors", "if-contains-scalable-vectors",
+           "bool", /*default=*/"false",
+           "Apply the selected streaming/ZA modes if the function contains "
+           "operations that use scalable vector types.">
```

Comment:
The distinction between ought to be clearer. Here's a suggestion:

**ifRequredByOps**
> Apply the selected streaming/ZA modes if the function contains ops that require them (e.g. don't enable ZA, even when requested, when there are no ops operating on 2d scalable vectors).

**ifContainsScalableVectors"**
> Apply the selected streaming/ZA modes if the function contains operations that use scalable vector types (e.g. don't enable SSVE and/or ZA, even when requested, when there are no scalable vectors).

WDYT? (I've only really added an example)

Reply:
I've tried to improve the docs. I find the suggested wording a little confusing though :sweat_smile: (mainly the "even when requested" since these flags make the "request" conditional).